app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,33 @@
+module SessionsHelper
+
+  def log_in(user)
+    reset_session
+    session[:user_id] = user.id
+    session[:cart] = Hash.new(0)
+  end
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+  end
+
+  def logged_in?
+    !current_user.nil?
```

Comment:
I would use here `current_user.present?` instead of using negation. It is much more readable from my perspective

Reply:
Thanks!